lib/repo/git_hooks/client/pre-commit.d/02-block_change_top_level.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env sh
+
+echo '[MarkUs] Checking whether top-level files/directories were created/modified...'
+
+CHANGED_FILES=$(git diff-index --name-only HEAD)
+
+if printf '%s' "$CHANGED_FILES" | grep -qv -e '/' -e '.gitignore'; then
+  echo "[MarkUs] Error: Top-level change detected for the following entries:"
+  echo "$CHANGED_FILES" | grep -v -e '/' -e '.gitignore'
+  echo "[MarkUs] You should unstage this change."
```

Comment:
Okay this is a bit picky, but I realized that this hook checks for changes to top-level files regardless of whether they're staged or not.

While my preference would be to modify the hook to only check for staged files, I believe this was the previous behaviour (since you're using the same `git diff-index` as the Python version). So I'd also be okay keeping the behaviour but rewording this specific error message to say `"You should undo this change."` rather than `"You should unstage this change."`